plan/physical_plan_builder.go

```diff
@@ -89,9 +89,18 @@ func (p *baseLogicalPlan) findBestTask(prop *requiredProp) (bestTask task, err e
 		return invalidTask, nil
 	}
 
+	// If prop.enforced is true, cols of prop as parameter in exhaustPhysicalPlans should be nil
+	// And reset it for enforcing task prop and storing map<prop,task>
+	oldPropCols := prop.cols
+	if prop.enforced {
+		prop.cols = []*expression.Column{}
```

Comment:
consider this:
```sql
create table t1(a bigint, b bigint, index idx_1(a));
create table t2(a bigint, b bigint);
select /*+ TIDB_SMJ(t1, t2) */ * from t1 join t2 on t1.a=t2.a;
```

```sql
TiDB(localhost:4000) > desc select /*+ TIDB_SMJ(t1, t2) */ * from t1 join t2 on t1.a=t2.a;
+----------------+-------------+----------------+------+---------------------------------------------------------------------------------------+----------+
| id             | parents     | children       | task | operator info                                                                         | count    |
+----------------+-------------+----------------+------+---------------------------------------------------------------------------------------+----------+
| TableScan_7    |             |                | cop  | table:t1, range:[-inf,+inf], keep order:false                                         | 10000.00 |
| TableReader_8  | Sort_9      |                | root | data:TableScan_7                                                                      | 10000.00 |
| Sort_9         | MergeJoin_6 | TableReader_8  | root | test.t1.a:asc                                                                         | 10000.00 |
| TableScan_10   |             |                | cop  | table:t2, range:[-inf,+inf], keep order:false                                         | 10000.00 |
| TableReader_11 | Sort_12     |                | root | data:TableScan_10                                                                     | 10000.00 |
| Sort_12        | MergeJoin_6 | TableReader_11 | root | test.t2.a:asc                                                                         | 10000.00 |
| MergeJoin_6    |             | Sort_9,Sort_12 | root | inner join, equal:[eq(test.t1.a, test.t2.a)], left key:test.t1.a, right key:test.t2.a | 12500.00 |
+----------------+-------------+----------------+------+---------------------------------------------------------------------------------------+----------+
7 rows in set (0.00 sec)
```

actually we can use index `idx_1` on `t1` to avoid a `Sort_9 `